StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Vertical Main Screen/AccessoryButton.swift

```diff
@@ -0,0 +1,85 @@
+//
+//  AccessoryButton.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 5/28/24.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+@_spi(STP) import StripeUICore
+import UIKit
+
+final class AccessoryButton: UIButton {
+
+    enum AccessoryType {
+        case none
+        case edit
+        case viewMore
+
+        var text: String? {
+            switch self {
+            case .none:
+                return nil
+            case .edit:
+                return .Localized.edit
+            case .viewMore:
+                return .Localized.view_more
+            }
+        }
+
+        var accessoryImage: UIImage? {
+            switch self {
+            case .none, .edit:
+                return nil
+            case .viewMore:
+                return Image.icon_chevron_right.makeImage(template: true)
+            }
+        }
+
+        var imageEdgeInsets: UIEdgeInsets {
+            switch self {
+            case .none, .edit:
+                return .zero
+            case .viewMore:
+                return UIEdgeInsets(top: 2, left: 4, bottom: 0, right: 0)
+            }
+        }
+    }
+
+    override var intrinsicContentSize: CGSize {
+        let size = super.intrinsicContentSize
+
+        // TODO(porter) Figure out how to handle this for VisionOS, the alternative (`UIButton.Configuration) was introduced in iOS 15 and we currently support below iOS 15.
```

Comment:
What are we trying to solve by overriding this method?

Reply:
Overriding this ensures that auto layout respects the custom padding to accurately size and position the button. Without this, if we move the image to align with the figmas it could draw outside of it's size and that doesn't mesh well with auto layout. So by taking into account the image edge sets we can define the size more accurately. I added a comment.